Mathlib/Order/SuccPred/Limit.lean

```diff
@@ -67,33 +65,71 @@ section Preorder
 
 variable [Preorder α] {a : α}
 
+/-- A successor limit is a value that isn't minimal and doesn't cover any other.
+
+It's so named because in a successor order, a successor limit can't be the successor of anything
+smaller.
+
+This previously allowed the element to be minimal. This usage is now covered by `IsSuccPrelimit`. -/
+def IsSuccLimit (a : α) : Prop :=
+  ¬ IsMin a ∧ IsSuccPrelimit a
+
+protected theorem IsSuccLimit.not_isMin (h : IsSuccLimit a) : ¬ IsMin a := h.1
+protected theorem IsSuccLimit.isSuccPrelimit (h : IsSuccLimit a) : IsSuccPrelimit a := h.2
+
+theorem IsSuccPrelimit.isSuccLimit_of_not_isMin (h : IsSuccPrelimit a) (ha : ¬ IsMin a) :
+    IsSuccLimit a :=
+  ⟨ha, h⟩
+
+theorem IsSuccPrelimit.isSuccLimit [NoMinOrder α] (h : IsSuccPrelimit a) : IsSuccLimit a :=
+  h.isSuccLimit_of_not_isMin (not_isMin a)
+
+theorem isSuccPrelimit_iff_isSuccLimit_of_not_isMin (h : ¬ IsMin a) :
+    IsSuccPrelimit a ↔ IsSuccLimit a :=
+  ⟨fun ha ↦ ha.isSuccLimit_of_not_isMin h, IsSuccLimit.isSuccPrelimit⟩
+
+theorem isSuccPrelimit_iff_isSuccLimit [NoMinOrder α] : IsSuccPrelimit a ↔ IsSuccLimit a :=
```

Comment:
Should this be simp?

Reply:
I don't think so; on the one hand `IsSuccLimit` provides strictly more information, but on the other, `IsSuccPrelimit` has a larger API.

Reply:
I don't entirely agree, but that can be changed later